NodeProg/Infrastructure/NumberNode-TryParseInt64.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NodeProg
+{
+    public partial class NumberNode : Node
+    {
+        private static long? TryParseInt64(
+            string input,
+            int startIndex,
+            int length,
+            bool allowLeadingWhitespace = true,
+            bool allowTrailingWhitespace = true)
+        {
+            if (length == 0)
+                return null;
+
+            var position = startIndex;
+
+
+            if (allowLeadingWhitespace)
+            {
+                while (position < startIndex + length)
+                {
+                    if (!(input[position] == ' ' || '\t' == input[position]))
```

Comment:
try to not duplicate the same access expressions like `input[position]`, cache it in a variable.